homeassistant/components/modbus/validators.py

```diff
@@ -243,6 +251,51 @@ def duplicate_fan_mode_validator(config: dict[str, Any]) -> dict:
     return config
 
 
+def ensure_and_check_conflicting_scales_and_offsets(config: dict[str, Any]) -> dict:
+    """Check for conflicts in scale/offset and ensure target/current temp scale/offset is set."""
+
+    if CONF_TARGET_TEMP_SCALE not in config:
+        config[CONF_TARGET_TEMP_SCALE] = config.get(CONF_SCALE, DEFAULT_SCALE)
+    elif CONF_SCALE in config and config[CONF_TARGET_TEMP_SCALE] != config[CONF_SCALE]:
```

Comment:
No !!!! scale and target/current are not to be used simultaneously.

Reply:
This goes right down all the checks.

This method could be simplified quite a lot.